homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -244,6 +244,24 @@ async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowRes
             )
         )
 
+        def _is_supported_device() -> bool:
+            """
+            See if we are looking at a possibly supported device.
+
+            Matching solely on SSDP data does not yield reliable enough results.
+            """
+            try:
+                with Connection(url=url, timeout=CONNECTION_TIMEOUT) as conn:
+                    basic_info = Client(conn).device.basic_information()
+            except ResponseErrorException:  # API compatible error
+                return True
+            except Exception:  # API incompatible error # pylint: disable=broad-except
+                return False
+            return isinstance(basic_info, dict)  # Crude content check
+
+        if not await self.hass.async_add_executor_job(_is_supported_device):
+            return self.async_abort(reason="unsupported_device")
```

Comment:
I'm not sure how the abort works, but does this create an unnecessary notification on false positives instead of simply ignoring it for future ssdp discoveries?

Reply:
Aborting here does not result in anything user visible by default; no notification, not even log entries visible with the default logging settings.

The unique id machinery takes care of preventing another discovery instance happening while one for the unique id is in progress, has been completed, or has been set as ignored by the user. With 097f834295016b5bd23967640036db43b013bea4 that actually takes place before we probe, thanks for the sanity check!

If a discovery is not acted on and HA is restarted, then there will be another probe after start, i.e. the unique id prevention measures for inflight discoveries are not persistent. But I think that's fine.